### PR TITLE
Sup 37 get warehouse info

### PR DESCRIPTION
**Pull request by alexeyoverchenko**
for review

### CODE REVIEW

File: src/main/java/by/itech/lab/supplier/controller/WarehouseController.java [hunk 1]
```diff
@@ -1,13 +1,37 @@
...
 @AllArgsConstructor
 @RequestMapping(URL_WAREHOUSE)
 public class WarehouseController {
+
+    private final WarehouseService warehouseService;
+
+    @GetMapping
+    public Page<WarehouseDto> findAllByActive(
+            @PageableDefault(sort = {"id"}, direction = Sort.Direction.DESC) final Pageable pageable,
+            @RequestParam(required = false) final Boolean status) {
```
Comment (svareyko) [inline src/main/java/by/itech/lab/supplier/controller/WarehouseController.java]: maybe would be simpler rename to 'active'?

File: src/main/java/by/itech/lab/supplier/repository/UserRepository.java [hunk 2]
```diff
@@ -16,7 +16,7 @@
     Optional<User> findById(Long id);
 
     @Modifying
-    @Query("update User set activeForDeletion = false where id = :id")
+    @Query("update User set deleted = false where id = :id")
```
Comment (svareyko) [inline src/main/java/by/itech/lab/supplier/repository/UserRepository.java]: from the logic side: if `deleted == true` - its not deleted, but if `false` - its deleted. is it ok?

File: src/main/java/by/itech/lab/supplier/repository/WarehouseRepository.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package by.itech.lab.supplier.repository;
+
+import by.itech.lab.supplier.domain.Warehouse;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface WarehouseRepository extends JpaRepository<Warehouse, Long> {
```
Comment (svareyko) [inline src/main/java/by/itech/lab/supplier/repository/WarehouseRepository.java:8]: Should it be marked as `@Repository?`

File: src/main/java/by/itech/lab/supplier/service/impl/WarehouseServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+    @Override
+    public WarehouseDto findById(Long warehouseId) {
+        return warehouseRepository.findById(warehouseId).map(warehouseMapper::map)
+                .orElseThrow(NotFoundInDBException::new);
+    }
+
+    @Override
+    public WarehouseDto save(WarehouseDto dto) {
+        return null;
```
Comment (svareyko) [inline src/main/java/by/itech/lab/supplier/service/impl/WarehouseServiceImpl.java:34]: ?

**Review by svareyko — CHANGES_REQUESTED** [review]

File: src/main/java/by/itech/lab/supplier/controller/WarehouseController.java (hunk 1 above)
Comment (alexeyoverchenko) [reply src/main/java/by/itech/lab/supplier/controller/WarehouseController.java]: it's name from BaseServise interface. we use the same names

File: src/main/java/by/itech/lab/supplier/repository/UserRepository.java (hunk 2 above)
Comment (alexeyoverchenko) [reply src/main/java/by/itech/lab/supplier/repository/UserRepository.java]: Yes, we all should fix it in our classes. we'll talk about it tomorrow

File: src/main/java/by/itech/lab/supplier/repository/WarehouseRepository.java (hunk 3 above)
Comment (alexeyoverchenko) [reply src/main/java/by/itech/lab/supplier/repository/WarehouseRepository.java:8]: done

File: src/main/java/by/itech/lab/supplier/service/impl/WarehouseServiceImpl.java (hunk 4 above)
Comment (alexeyoverchenko) [reply src/main/java/by/itech/lab/supplier/service/impl/WarehouseServiceImpl.java:34]: I've implemented BaseServer methods. "save" will need it to the future

File: src/main/java/by/itech/lab/supplier/controller/WarehouseController.java (hunk 1 above)
Comment (svareyko) [reply src/main/java/by/itech/lab/supplier/controller/WarehouseController.java]: maybe need to rename there too?

File: src/main/java/by/itech/lab/supplier/repository/UserRepository.java (hunk 2 above)
Comment (svareyko) [reply src/main/java/by/itech/lab/supplier/repository/UserRepository.java]: as per our discussion everyone who will change something near to places where its should be renamed - should rename it

File: src/main/java/by/itech/lab/supplier/repository/WarehouseRepository.java (hunk 3 above)
Comment (svareyko) [reply src/main/java/by/itech/lab/supplier/repository/WarehouseRepository.java:8]: I cant see you "done" until you have not uploaded it here

File: src/main/java/by/itech/lab/supplier/service/impl/WarehouseServiceImpl.java (hunk 4 above)
Comment (svareyko) [reply src/main/java/by/itech/lab/supplier/service/impl/WarehouseServiceImpl.java:34]: so, mark as TODO

**Review by svareyko — APPROVED** [review]
Please rename status to active in the controller and then it can be merged

File: src/main/java/by/itech/lab/supplier/service/impl/WarehouseServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+@Service
+@AllArgsConstructor
+public class WarehouseServiceImpl implements WarehouseService {
+
+    private final WarehouseRepository warehouseRepository;
+    private final WarehouseMapper warehouseMapper;
+
+    @Override
+    public Page<WarehouseDto> findAllByActive(Pageable pageable, Boolean active) {
+        return warehouseRepository.findAll(pageable).map(warehouseMapper::map);
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/service/impl/WarehouseServiceImpl.java:22]: According to TR  Warehouses can't be enable or disable

Comment (alexeyoverchenko) [reply src/main/java/by/itech/lab/supplier/service/impl/WarehouseServiceImpl.java:22]: I've corrected it

File: src/main/java/by/itech/lab/supplier/dto/mapper/WarehouseMapper.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+import by.itech.lab.supplier.domain.Warehouse;
+import by.itech.lab.supplier.dto.WarehouseDto;
+import lombok.AllArgsConstructor;
+
+import java.util.stream.Collectors;
+
+@AllArgsConstructor
+public class WarehouseMapper implements BaseMapper<Warehouse, WarehouseDto> {
+
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/dto/mapper/WarehouseMapper.java:13]: Maybe you lost @Component

File: src/main/java/by/itech/lab/supplier/domain/Warehouse.java [hunk 7]
```diff
@@ -14,83 +25,27 @@
 @NoArgsConstructor
 @Builder
 @Entity
-@Table(name = "warehouse")
-public class Warehouse {
+@Table
+@Where(clause = "deleted_at is null")
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/domain/Warehouse.java]: You don't realize such functionality, remove it

File: src/main/java/by/itech/lab/supplier/service/impl/CustomerServiceImpl.java [hunk 8]
```diff
@@ -24,7 +24,7 @@
     private final UserServiceImpl userService;
 
     @Override
-    public Page<CustomerDto> findAllByActive(final Pageable pageable, final Boolean status) {
+    public Page<CustomerDto> findAll(final Pageable pageable, final Boolean status) {
         return customerRepository.findByStatus(pageable, status).map(customerMapper::mapToCustomerView);
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/service/impl/CustomerServiceImpl.java]: I don't understand why you rename other services

File: src/main/java/by/itech/lab/supplier/service/impl/WarehouseServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    @Override
+    public Page<WarehouseDto> findAll(Pageable pageable) {
+        return warehouseRepository.findAll(pageable).map(warehouseMapper::map);
+    }
+
+    @Override
+    public WarehouseDto findById(Long warehouseId) {
+        return warehouseRepository.findById(warehouseId).map(warehouseMapper::map)
+                .orElseThrow(NotFoundInDBException::new);
+    }
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/service/impl/WarehouseServiceImpl.java:29]: If I try to get a resource which doesn't exist I'm getting 500 error. You throw an exception but never handling it

File: src/main/java/by/itech/lab/supplier/service/impl/WarehouseServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+    // TODO: 11/4/20
+    @Override
+    public WarehouseDto save(WarehouseDto dto) {
+        return null;
+    }
+
+    @Override
+    public Page<WarehouseDto> findAllByActive(Pageable pageable, Boolean active) {
+        return null;
+    }
```
Comment (IrynaZazyba) [inline src/main/java/by/itech/lab/supplier/service/impl/WarehouseServiceImpl.java]: null?

**Review by IrynaZazyba — APPROVED** [review]

**Pull request merged**